### ISSUE TITLE
Backspace and enter lag due to conflicts in recommended extensions (spellright and vscode-markdown)

### ISSUE THREAD

**Issue by j-mesnil** [foambubble/foam#1068]
### Describe the bug

There is a [known bug](https://github.com/yzhang-gh/vscode-markdown/issues/423) that introduces high amounts of lag when editing markdown files when both the spellright and the vscode-markdown extensions are enabled. Both of these extensions are recommended to use with foam, and I encountered the lag soon after starting to use foam.

The lag occurs typically when the enter-, tab-, or backspace-keys are hit, and the effect of the lag is that the corresponding keystroke is not executed sometimes for several seconds. In my case, one effect is that if I want to delete whole words using the backspace key in a markdown file, only the first back-space keystroke is recognized, after which no other backspace keystroke is recognized for several seconds even.

There is also a [known workaround](https://github.com/yzhang-gh/vscode-markdown/issues/423#issuecomment-610508023) around this problem, which involves deactivating some keyboard shortcuts associated with the vscode-markdown extension. This workaround worked for me (in my case, deactivating only the shortcut associated with backspace proved sufficient for a good UX).

I am wondering if there is some way for foam to set a configuration for the vscode-markdown extension that disables this keyboard shortcut by default, or if there may be some other way to circumvent this problem for new foam users.

### Steps to Reproduce the Bug or Issue

1. Install foam according to the [getting started instructions](https://foambubble.github.io/foam/#getting-started), while installing the recommended extensions.
2. Set up the spellright extension to check for spelling errors in markdown documents.
3. In a markdown document in foam, edit some text.
4. Using the backspace while editing results in significant lag in the editor.

### Expected behavior

When using the backspace while writing in a markdown file, I expect to experience no or minimal lag so that also consecutive backspace usages are handled smoothly.

### Operating System Version

Linux, Ubuntu 20.04.5 LTS

### Visual Studio Code Version

1.71.0

**Comment by riccardoferretti** [comment]
Thanks for this @j-mesnil 
Here is my recommended solution:

1. remove spellcheck from the list of recommended extension in the documentation and in `foam-template`: there have been a few reports of issues with this extension, and it doesn't need to be part of the core Foam distribution
2. add spellcheck as a recipe, with the information you shared in this ticket so that users know what they might encounter and easily debug/fix the problem

What do you think?

**Comment by j-mesnil** [comment]
@riccardoferretti I think that could be a sensible solution - I don't know about the other problems caused by the spellchecker extension so I'm not qualified to judge them, but it would solve the problem I described.

I might get around to write such a recipe in the next days, but I am not yet entirely sure about that. Will see :)

**Comment by alexzanutto** [comment]
@j-mesnil @riccardoferretti I am using [LanguageTool Linter](https://github.com/davidlday/vscode-languagetool-linter) instead of Spell Right. Now, I am not sure that they do exactly the same thing (I am not a dev and I just started using VSCode to use Foam), but I am assuming that if Spell Right is used to catch spelling mistakes, then LanguageTool could be a good alternative. 

As I said I am no dev, and it took a bit of trial and error to set up the extension in the recommended configuration with Docker, but in the end, it seems to be working okay, and I have found no backspace lagging.

Let me know if this could be a good idea to be developed into a recipe or something (still figuring out how to contribute to Foam)
